src/route/Params.tsx

```diff
@@ -1,72 +1,454 @@
-import React from "react";
-import { map, assign } from "lodash-es";
+import React, { useEffect } from "react";
+import { map, startCase } from "lodash-es";
 
 import Input from "../common/Input";
 import Label from "../common/Label";
+import Required from "../common/Required";
+import Button from "../common/Button";
+
 import Helpers from "../lib/helpers";
 
-import { Route, Param, ParamType } from "../types";
+import { useGlobalContext } from "../context/Context";
+
+import { Route, Param, ParamType, Size } from "../types";
 
-type RefetchFn = () => void;
-interface MapInputsOptions {
+type OnSubmitFn = () => void;
+interface RenderInputs {
+  pathToValue: string;
   inputs: Param[];
-  route: Route;
-  paramType: ParamType;
-  reFetch: RefetchFn;
-  parent: null | string;
+  onSubmit: OnSubmitFn;
+  className: string;
 }
 
-function mapInputs({
-  inputs,
-  route,
-  paramType,
-  reFetch,
-  parent,
-}: MapInputsOptions) {
-  return map(inputs, (param: Param, idx: number) => {
-    // Object datatype
-    if (param.properties) {
-      return (
-        <div className="mb-2" key={idx}>
-          <Label size="lg">{param.label}</Label>
-          {mapInputs({
-            inputs: param.properties,
-            route,
-            paramType,
-            reFetch,
-            parent: param.name,
-          })}
-        </div>
-      );
+interface RenderInputByDataType {
```

Comment:
Though this isn't a problem, I would consider doing this for all the props interfaces.

```suggestion
interface RenderInputByDataTypeProps {
```